biz.aQute.bndlib/src/aQute/bnd/build/Project.java

```diff
@@ -270,7 +270,7 @@ public void prepare() throws Exception {
 			return;
 		}
 
-		synchronized (preparedPaths) {
+		synchronized (this) {
```

Comment:
I thought we did not agree to do this and closed the issue?

Reply:
ok

Reply:
Then we should  reopen the issue you created for this and continue the discussion.

Reply:
I found it slightly safer to lock on the Project instead of the preparedPaths because other places also lock on the project, which caused the original deadlock.

Reply:
This is a minute change?

Reply:
It is a change that we disagree about as discussed at length in https://github.com/bndtools/bnd/pull/3455 and, at the bndtools call, agreed to close the PR and leave things as they are. This seems to ignore all that and try and slide the change in as part of this very large PR.

Reply:
discussion was as I recall to remove the sync form Bndtools?

Reply:
but I don't care enough about. I reverted although I think in this case locking on Project is better

Reply:
I believe so. Nowhere is it defined that one should sync on Project or what value that would provide. So we should examine all place which sync on Project and remove and possible replace with something else if actually necessary.